### PR TITLE
[feature] Unify Extra Trees and Random Forest preview classes

### PR DESCRIPTION
**Pull request by icfaust**
# Description
This unifies Random Forest and Extra Trees to a single file from the same inheritance classes to match stock scikit-learn. This reduces overall code size, and prepares for movement out of preview.  This improves preview RF to meet some of the standards of sklearnex in terms of testing and verbosity.

Changes proposed in this pull request:
- rm extra_trees.py
- use extra_trees.py code as base for forest.py
- set Random Forest Classification and Random Forest Regression as child classes of ForestClassifier and ForestRegressor

### CODE REVIEW

**Comment by icfaust** [comment]
Todos: Fix testing issue, make changes for v1.4 (monotonic_cst) that were only made to forest.py

File: sklearnex/preview/ensemble/forest.py [hunk 1]
```diff
@@ -36,26 +39,34 @@
...
 
 if sklearn_check_version("1.2"):
-    from sklearn.utils._param_validation import Interval, StrOptions
-if sklearn_check_version("1.4"):
-    from daal4py.sklearn.utils import _assert_all_finite
+    from sklearn.utils._param_validation import Interval
 
 
-class BaseRandomForest(ABC):
+class BaseTree(ABC):
```
Comment (samir-nasibli) [inline sklearnex/preview/ensemble/forest.py:69]:
```suggestion
class BaseForest(ABC):
```

File: sklearnex/preview/ensemble/forest.py [hunk 2]
```diff
@@ -211,268 +214,103 @@ def estimators_(self, estimators):
...
+        estimator_params=tuple(),
+        bootstrap=False,
+        oob_score=False,
+        n_jobs=None,
+        random_state=None,
+        verbose=0,
+        warm_start=False,
+        class_weight=None,
+        max_samples=None,
+    ):
```
Comment (samir-nasibli) [inline sklearnex/preview/ensemble/forest.py:237]: 👍

**Review by samir-nasibli — COMMENTED** [review]
partially reviewed

**Pull request closed without merging**